src/mixer/basetrackplayer.cpp

```diff
@@ -415,8 +416,17 @@ void BaseTrackPlayerImpl::slotLoadFailed(TrackPointer pTrack, const QString& rea
         qDebug() << "Failed to load track (NULL track object)" << reason;
     }
     m_pChannelToCloneFrom = nullptr;
+
     // Alert user.
+    // Show only one message for this track at a time
+    TrackId currTrackId = pTrack->getId();
+    if (pTrack && m_pPrevFailedTrackId == currTrackId) {
```

Comment:
if pTrack is null, it has already crashed above. How about moving the `if (pTrack) {` to the beginning?

Reply:
> if pTrack is null, it has already crashed above.

I don't understand where it would have "crashed above"?
IMO that is irrelevant here because the existing code expects having to deal with this and would do `m_pChannelToCloneFrom = nullptr;`, and I have no intention to change that.

Reply:
The line:
`TrackId currTrackId = pTrack->getId();`
will segfault if pTrack is null 
so you will never reach the line below  
`if (pTrack &&`

Reply:
:man_facepalming:

Reply:
fixed